src/sdk/svc/node/list_registers_client.hpp

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+//
+
+#ifndef OCVSMD_SDK_SVC_NODE_LIST_REGISTERS_CLIENT_HPP_INCLUDED
+#define OCVSMD_SDK_SVC_NODE_LIST_REGISTERS_CLIENT_HPP_INCLUDED
+
+#include "ipc/client_router.hpp"
+#include "ocvsmd/sdk/node_registry_client.hpp"
+#include "svc/node/list_registers_spec.hpp"
+
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+
+namespace ocvsmd
+{
+namespace sdk
+{
+namespace svc
+{
+namespace node
+{
+
+/// Defines interface of the 'Node: List Registers' service client.
+///
+class ListRegistersClient
+{
+public:
+    using Ptr  = std::shared_ptr<ListRegistersClient>;
+    using Spec = common::svc::node::ListRegistersSpec;
+
+    using Result  = NodeRegistryClient::List::Result;
+    using Success = NodeRegistryClient::List::Success;
+    using Failure = NodeRegistryClient::List::Failure;
+
+    CETL_NODISCARD static Ptr make(cetl::pmr::memory_resource&           memory,
```

Comment:
shouldn't this take an allocator as the second argument with `allocator_arg_t` as the first argument (per [the leading-allocator convention](https://en.cppreference.com/w/cpp/memory/uses_allocator#Uses-allocator_construction))

Reply:
Sorry, I don't understand why/how some convention applies to my PRIVATE (not part of PUBLIC SDK includes) IPC implementation. Anyway, now I see that `memory` is not in use at all in this class (copy/paste from `ExecCmdClient` bug) - will delete it.

Reply:
BTW, the only reason why I need `cetl::pmr::memory_resource` on CLIENT side of SDK is b/c of Nunavut de/serialization generated code.